### PR TITLE
implement no_parent_owners

### PR DESCRIPTION
**Pull request by BenTheElder**
fixes https://github.com/kubernetes/test-infra/issues/5197

this will allow OWNERS files to look like:
```yaml
options:
  no_parent_owners: true
reviewers:
  - mml
approvers:
  - mml
```
Which will ensure that we do not search for OWNERS in the directory above (see the above issue).

### CODE REVIEW

**Comment by BenTheElder** [comment]
/area prow
Needs test cases, also not sure if this is the cleanest solution 🤷‍♂️

**Comment by cjwagner** [comment]
This is exactly what I was thinking of, I think its a clean solution given how the OWNERS info is stored.

**Comment by BenTheElder** [comment]
I updated the unit tests.

**Comment by cjwagner** [comment]
You need to check that the expected options match the actual loaded options in `TestLoadRepoOwners`.
Please also clean up the comments at the end of that function (I left them there by mistake). 😝

**Comment by BenTheElder** [comment]
@cjwagner done

File: prow/repoowners/repoowners.go [hunk 1]
```diff
@@ -338,6 +347,9 @@ func (o *RepoOwners) applyConfigToPath(path string, config *ownersConfig) {
 	if len(config.Labels) > 0 {
 		o.labels[path] = sets.NewString(config.Labels...)
 	}
+	if config.Options != defaultDirOptions {
```
Comment (BenTheElder) [inline prow/repoowners/repoowners.go]: fixing the unit test made me realize that I was also storing a `dirOptions` for every dir even if they were unset, hence this. :shrug:

**Comment by BenTheElder** [comment]
/test pull-test-infra-bazel-canary
🐦

**Pull request merged**